### PR TITLE
f32/f64: add InterleaveN/DeinterleaveN N-stream scatter/gather

### LINKED ISSUE #53: f32/f64: add InterleaveN / DeinterleaveN (N-stream scatter/gather)
## Summary

Generalize `Interleave2` / `Deinterleave2` to N streams in `f32` and `f64`:

```go
// InterleaveN: dst[i*N + c] = srcs[c][i],  N = len(srcs)
func InterleaveN(dst []float32, srcs [][]float32)

// DeinterleaveN: dsts[c][i] = src[i*N + c],  N = len(dsts)
func DeinterleaveN(dsts [][]float32, src []float32)
```

## Motivation

Only the 2-stream variants exist (`f32/f32.go:224`, `:235`). go-audio-resampler's integer upsample by factor > 2 combines phase outputs with a scalar nested loop (`DFTStage.processChunk` / `processChunked`, the `for phase := range factor` scatter in `internal/engine/dft_stage.go:265` and `:329`).

Integer upsamples like 16 kHz -> 48 kHz (3x) and 8 kHz -> 48 kHz (6x) hit exactly this scalar path on the float32 streaming pipeline (low-rate sources upsampled to 48 kHz are common). The same primitive also serves planar <-> interleaved multichannel audio in general.

## Semantics

- N is derived from `len(srcs)` / `len(dsts)`. N == 1 is a copy; N == 2 must match `Interleave2` / `Deinterleave2` exactly.
- Process `min` over the per-stream lengths and the interleaved length.

## Implementation notes

- Special-case small N (2, 3, 4, 6, 8) with register shuffles (a small transpose); generic stride fallback for arbitrary N.
- Avoid AVX-512 gather/scatter for small N (they are slow); prefer shuffle-based transposes.
- Allocation-free; caller-owned buffers.

## Tests and acceptance

- Parity vs scalar for N in {1,2,3,4,5,6,8}, assorted lengths and ragged tails; N == 2 matches the existing ops.
- `asmcheck_test.go` entries for new asm; benchmark vs the scalar nested loop.
- Allocation-free.

Scoped to a single PR. Planning only; implementation in a separate session.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Generalizes `Interleave2`/`Deinterleave2` to N planar streams in `f32` and `f64`, the planar <-> interleaved primitive that go-audio-resampler's factor > 2 integer upsample (16k -> 48k, 8k -> 48k) currently does in a scalar nested loop.

```go
func InterleaveN(dst []float32, srcs [][]float32)   // dst[i*N + c] = srcs[c][i]
func DeinterleaveN(dsts [][]float32, src []float32) // dsts[c][i] = src[i*N + c]
```

N is derived from `len(srcs)`/`len(dsts)`. N == 1 is a copy; N == 2 produces the same result as `Interleave2`/`Deinterleave2`. The frame count clamps to `min(len(dst)/N, shortest stream)`; ragged tails and `dst` past the clamp are left untouched. Allocation-free, caller-owned buffers, preserving the float32 zero-alloc streaming contract.

## SIMD coverage

| N | f32 AMD64 | f64 AMD64 | f32/f64 ARM64 |
|---|-----------|-----------|---------------|
| 2 | AVX (existing) | AVX (existing) | NEON ZIP/UZP (existing) |
| 3 | generic Go | generic Go | **NEON ST3/LD3** |
| 4 | **AVX 4x4 transpose** | **AVX 4x4 transpose** | **NEON ST4/LD4** |
| 8 | **AVX 8x8 transpose** | generic Go | generic Go |
| other | generic Go | generic Go | generic Go |

ARM64's structured load/store (ST3/LD3/ST4/LD4) maps 1:1 to the operation, so it covers the hot N=3 path directly. On a Raspberry Pi 5, N=3 interleave runs about 10x the scalar path (770 ns vs 8105 ns for 1024 frames) and deinterleave about 13x. On x86 the N=4/N=8 transposes run about 6-7x scalar. Everything not specialized uses the allocation-free generic Go path.

The remaining shuffle-based cells (AMD64 f32 N=3/N=6 and f64 N=3/N=6/N=8) need bespoke `VPERMPS`/`VPERMPD` gather-blend or multi-register transposes for a more modest x86 gain; they are tracked as a follow-up to be gated on x86 profiling rather than landed speculatively.

## Tests

`interleave_n_test.go` in each package: parity against an independent scalar oracle for N in {1,2,3,4,5,6,8} across assorted lengths and ragged tails (spanning every SIMD block boundary and its +-1 tail), N == 2 bit-equivalence with `Interleave2`/`Deinterleave2`, round-trip identity, length clamping, empty-input guards, and `AllocsPerRun == 0`. Benchmarks vs the scalar nested loop included.

Verified on amd64 and on a Raspberry Pi 5 (arm64). `golangci-lint` clean on both architectures; the ARM64 WORD-encoding cross-check still passes (the new structured load/stores use Go assembler mnemonics, not hand-encoded `WORD`s).

Closes #53. Part of #54.

## Summary by CodeRabbit

* **New Features**
  * Added generalized multi-channel audio interleaving functions supporting arbitrary channel counts (3, 4, and beyond) for 32-bit and 64-bit floating-point data, complementing existing 2-channel support with allocation-free streaming behavior.

* **Tests**
  * Added comprehensive test suites and performance benchmarks validating multi-channel interleaving across various channel counts and edge cases.

### CODE REVIEW

File: f32/f32_arm64.s [hunk 1]
```diff
@@ -649,6 +649,184 @@ deinterleave2_neon32_loop1:
...
+    FMOVS (R3), F2
+    FMOVS F0, (R0)
+    FMOVS F1, 4(R0)
+    FMOVS F2, 8(R0)
+    ADD $4, R1
+    ADD $4, R2
+    ADD $4, R3
+    ADD $12, R0
+    SUB $1, R4
+    CBNZ R4, interleave3_neon_tail1
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/f32_arm64.s:689]: Intentional, leaving as-is. The WORD-encoding convention in this repo is for NEON ops the Go arm64 assembler cannot express as mnemonics (ZIP1/UZP1, vector FADD/FSUB, and similar). VLD1/VST1/VLD3/VST3/VLD4/VST4 and FMOVS are all natively supported by the Go assembler, which emits and validates the correct encodings at build time, the same way the existing interleave2NEON kernel uses native VLD1.P/VST1.P. There is no requirement to hand-encode these as WORD, and doing so would bypass the assembler validation for no benefit. asmcheck only cross-checks WORD directives; native mnemonics are covered by the assembler itself. These kernels build and pass the full parity, round-trip and zero-alloc suite on amd64 and on a Raspberry Pi 5 (arm64).

File: f64/f64_arm64.s [hunk 2]
```diff
@@ -726,6 +726,150 @@ deinterleave2_neon_remainder:
...
+    FMOVD 8(R4), F1
+    FMOVD 16(R4), F2
+    FMOVD 24(R4), F3
+    FMOVD F0, (R0)
+    FMOVD F1, (R1)
+    FMOVD F2, (R2)
+    FMOVD F3, (R3)
+
+deinterleave4_neon_done:
+    RET
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f64/f64_arm64.s:871]: Intentional, leaving as-is. The WORD-encoding convention in this repo is for NEON ops the Go arm64 assembler cannot express as mnemonics (ZIP1/UZP1, vector FADD/FSUB, and similar). VLD1/VST1/VLD3/VST3/VLD4/VST4 and FMOVS are all natively supported by the Go assembler, which emits and validates the correct encodings at build time, the same way the existing interleave2NEON kernel uses native VLD1.P/VST1.P. There is no requirement to hand-encode these as WORD, and doing so would bypass the assembler validation for no benefit. asmcheck only cross-checks WORD directives; native mnemonics are covered by the assembler itself. These kernels build and pass the full parity, round-trip and zero-alloc suite on amd64 and on a Raspberry Pi 5 (arm64).

File: README.md [hunk 3]
```diff
@@ -127,6 +127,8 @@ fmt.Println(cpu.HasFP16())     // true/false (ARM64 half-precision SIMD)
 |                 | `AccumulateAdd(dst, src, off)`      | Overlap-add: dst[off:] += src | 8x / 4x / 2x                        |
 | **Audio**       | `Interleave2(dst, a, b)`            | Pack stereo: [L,R,L,R,...]    | 4x / 2x                             |
 |                 | `Deinterleave2(a, b, src)`          | Unpack stereo to channels     | 4x / 2x                             |
+|                 | `InterleaveN(dst, srcs)`            | Pack N planar streams (N-stream Interleave2) | N=2,4 AVX / N=2,3,4 NEON |
+|                 | `DeinterleaveN(dsts, src)`          | Unpack N interleaved streams  | N=2,4 AVX / N=2,3,4 NEON            |
```
[the remark this answers is not in the document]
Comment (tphakala) [reply README.md]: Fixed in d8660e1: the InterleaveN and DeinterleaveN rows now note the allocation-free generic Go fallback for arbitrary N, in addition to the SIMD-specialized counts.
